gcloud/orb.yaml

```diff
@@ -252,6 +296,102 @@ jobs:
           command: cat output
           when: always
 
+  deploy-gke-TODO-NAME:
```

Comment:
`deploy-gke-app-w-autoscaler` ?
`deploy-gke-with-autoscal[er|ing]` ?

Reply:
Done, we'll go with `deploy-gke-with-autoscalering` :D

Reply:
Another option might be to see if we can merge this job with `deploy-gke`, if it's possible to write this such that the old, deprecated behaviour is the fallback case!

Reply:
I took a look at this option and I'm not sure it can fit without the job becoming a bit unwieldy. The part that concerns me is that it can operate in too many modes: `from_template` (for which you need to specify `template_path`), `image` (where you must _not_ specify `from_template` and `template_path`), and a new `from_yaml` (for which you must specify a `yaml_path`). Not to mention including `prune_selector` and `version_selector` which _cannot_ be sacrificed. I don't see a clean way to continue to use `deploy-gke` in a backwards-compatible way. Thoughts?

How about something like `deploy-gke-from-yaml`?

Reply:
Is the long term goal to replace `deploy-gke` with this new job? If so, I'd probably be in favour of just nuking the old job now and reusing the name, then doing a major version bump -- I'm mostly just worried about how many different ways we offer to do GKE deployments and how easy it might be to use the wrong one.

Also open to renaming/modifying the other job if that makes things easier -- eg. do we have any need for the `from_template` conditionals in the original `deploy-gke` job? We could strip those out and rename it to `bump-gke-image` or such?

Reply:
I could get behind doing a replacement -- I'll take a look at how we handle things in our repos at this point in time and figure out what the "refactor" would look like.

Reply:
Looks pretty straightforward -- let's go ahead with replacing the old job with this one.